### DISCUSSION TITLE
Adding Jar Dependencies instead of "DEPS"

### DISCUSSION THREAD

**Discussion by ShivKJ** [jbangdev/jbang#1810] [Q&A]
Hi,

Is there a way to include jar dependencies directly instead of providing them in `DEPS`?

For example, I am creating a `fat-jar` for my java project which has all the dependencies of the project in it. I see there is an option `--cp` but if the dependency is itself used in the `sourceOrFile` then it throws error.

```bash
jbang app install --cp target/template-maven-java-1.0-jar-with-dependencies.jar -o --force src/main/java/org/example/Main.java
```
In `Main.java` I am using/importing some of the dependencies present in `target/template-maven-java-1.0-jar-with-dependencies.jar` file.

Also once we have installed an app using `jbang app install ...`, is there a way to enable autocomplete for the app.

I am using **picocli** for command line application.

**Comment by maxandersen** [comment]
`--cp` is the answer to how to use .jar files as dependencies. What error are you getting?

> Also once we have installed an app using `jbang app install ...`, is there a way to enable autocomplete for the app.

That is up to your app - not related to jbang.

you mention you use picocli so I suggest you look at https://picocli.info/autocomplete.html to see examples on how to do it.

jbang itself use the same mechanics for its commands.

**Comment by ShivKJ** [comment]
Here is the error,

```bash
...
...

template-maven-java/src/main/java/org/example/Main.java:36: error: cannot find symbol
        int systemExit = new CommandLine(new Main()).execute(args);
                             ^
  symbol:   class CommandLine
  location: class Main
template-maven-java/src/main/java/org/example/Main.java:37: error: cannot find symbol
        System.out.println(Ints.max(1, 2, 5));
                           ^
  symbol:   variable Ints
  location: class Main

...
...
```

Top of my `Main.java` looks like,

```
// /usr/bin/env jbang "$0" "$@" ; exit $?
...
```

I ran the following command,

```bash
jbang app install --cp target/template-maven-java-1.0-jar-with-dependencies.jar  src/main/java/org/example/Main.java
```

Note that in `target/template-maven-java-1.0-jar-with-dependencies.jar`, `picocli` is added as one of the dependencies, along with other dependencies which is sees as ` error: cannot find symbol` in above error.
